src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -2721,7 +2713,10 @@ namespace SurfaceGeometry {
             }
             if (thisSurf.spaceNum > 0) {
                 anySurfacesWithSpace(thisSurf.Zone) = true;
-            } else {
+            } else if (thisSurf.ExtBoundCond != unreconciledZoneSurface) {
+                anySurfacesWithoutSpace(thisSurf.Zone) = true;
+            } else if (thisSurf.Name.substr(0, 3) != "iz-") {
+                // Only trigger a new space if the spaceless surface is not an autogenerated interzone surface
                 anySurfacesWithoutSpace(thisSurf.Zone) = true;
```

Comment:
Without breaking other situations where a surface has no space assigned, set the appropriate flags so that an autogenerated surface with the Zone BC gets assigned to an appropriate space.

Reply:
So you don't need an `else` block anymore at all?